controllers/machine_controller_test.go

```diff
@@ -29,16 +28,236 @@ import (
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/utils/pointer"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-var _ = Describe("Machine Reconciler", func() {
-	It("Should create a Machine", func() {
-		// TODO
-	})
-})
+func TestMachineFinalizer(t *testing.T) {
+	clusterCorrectMeta := &clusterv1.Cluster{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
Most kubernetes code & tests omit setting TypeMeta. It's ok to have it here, but it's also ok to omit it.

Reply:
Sounds good. I was thinking of omitting it myself to reduce unnecessary verbosity.